link-grammar/dict-common/print-dict.c

```diff
@@ -822,8 +822,8 @@ static char *display_disjuncts(Dictionary dict, const Dict_node *dn,
 		                                      max_cost, NULL);
 
 		unsigned int dnum0 = count_disjuncts(d);
-		d = eliminate_duplicate_disjuncts(d, false);
-		unsigned int dnum1 = count_disjuncts(d);
+		eliminate_duplicate_disjuncts(d, false);
+		unsigned int dnum1 = dnum0 - eliminate_duplicate_disjuncts(d, false);
 
```

Comment:
You caught a bug. The intention was to call it only once.
The idea here is to provide information on the number of duplicate disjuncts in the expression of the given word. With two calls, it would never report any duplicates. I will force-push a fix.

Without this fix:
``` text
linkparser> !!was.w-d//
Token "was.w-d" matches:
    was.w-d                        448 disjuncts


Token "was.w-d" disjuncts:
    was.w-d                        448/448 disjuncts

         was.w-d: [0] 1.000= @E- Ss- <> Xc+ Vv+ VC+
         was.w-d: [1] 0.000= @E- Ss- dIV- <> Xc+ Vv+ VC+
         was.w-d: [2] 0.000= @E- Ss- dIV- <> Xc+ Vv+ VC+ VC+
         ...
```
With the fix:
``` text
linkparser> !!was.w-d//
Token "was.w-d" matches:
    was.w-d                        448 disjuncts


Token "was.w-d" disjuncts:
    was.w-d                        352/448 disjuncts

         was.w-d: [0] 1.000= @E- Ss- <> Xc+ Vv+ VC+
         ...
```
Regarding the calls with `false` and `true,` this is done on a wildcard word to eliminate duplicate disjuncts with identical categories and then condense all the remaining duplicate disjuncts (disregarding the category). However, there is a code rot there, seemingly because I changed the implementation w/o fixing this code. It is inefficient, and when I look at this code again from a distance, I find nasty bugs, which may explain some strange things. I already rewrote part of the generation code for speed (and I know how to fix finding unused disjuncts), which I still need to continue. Instead, I'm now working on rewriting the counting stuff because I saw how to speed it up by a significant factor.

Reply:
> and when I look at this code again from a distance, I find nasty bugs, which may explain some strange things.

 To clarify, I referred to generation mode. I didn't find bugs regarding parsing mode. However, the speed can be improved (future PR).